src/components/currency.jsx

```diff
@@ -0,0 +1,23 @@
+var React = require('react');
+var valueMixin = require('./mixins/valueMixin.jsx');
+var multiMixin = require('./mixins/multiMixin.jsx');
+var inputMixin = require('./mixins/inputMixin.jsx');
+
+valueMixin.onChange =  function(e) {
```

Comment:
This is much closer. However, we need to either save of the original onChange function and then call it within the overridden onChange event or grab the index and call setValue. The reason is that fields can be multi value fields which means value may be an array. The best course of action is probably to call the original onChange event.